lib/Schema.ts

```diff
@@ -421,7 +421,8 @@ export class Schema {
 			}
 			const {typeDetails, matchedTypeDetailsIndex, matchedTypeDetailsIndexes} = typeCheckResult;
 			const hasMultipleTypes = Array.isArray(typeDetails);
-			const isObject = typeof value === "object" && value !== null;
+			// Added Fix for differentiating Buffer from rest of object types to avoid iterating over entries in Buffer
```

Comment:
```suggestion
```

Normally I'm a huge fan of code comments. But in this case the code is pretty self explanatory.

Reply:
Fixed in 4bb73de